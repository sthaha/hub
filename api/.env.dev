ENVIRONMENT="development"

POSTGRESQL_HOST="localhost"
POSTGRESQL_PORT="5432"
POSTGRESQL_DATABASE="hub"
POSTGRESQL_USER="postgres"
POSTGRESQL_PASSWORD="postgres"
