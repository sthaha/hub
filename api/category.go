package hub

import (
	"context"
	"log"

	category "github.com/tektoncd/hub/api/gen/category"
)

// category service example implementation.
// The example methods log the requests and return zero values.
type categorysrvc struct {
	logger *log.Logger
}

// NewCategory returns the category service implementation.
func NewCategory(logger *log.Logger) category.Service {
	return &categorysrvc{logger}
}

// Get all Categories with their associated tags.
func (s *categorysrvc) Categories(ctx context.Context) (res *category.CategoriesResult, err error) {
	res = &category.CategoriesResult{}
	s.logger.Print("category.Categories")
	return
}
