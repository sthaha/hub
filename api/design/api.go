package design

import (
	. "goa.design/goa/v3/dsl"
	cors "goa.design/plugins/v3/cors/dsl"
)

var _ = API("hub", func() {
	Title("Tekton Hub")
	Description("HTTP services for managing Tekton Hub")
	Version("1.0")
	Meta("swagger:example", "false")
	Server("hub", func() {
		Services("category", "swagger")
		Host("localhost", func() {
			URI("http://localhost:8000")
		})
	})
	cors.Origin("*", func() {
		cors.Headers("Content-Type")
		cors.Methods("GET")
	})
})
