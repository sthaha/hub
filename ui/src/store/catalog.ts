import { Instance, types } from 'mobx-state-tree';

export const Catalog = types
  .model({
    id: types.identifierNumber,
    name: types.optional(types.string, ''),
    type: types.optional(types.string, ''),
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalog = Instance<typeof Catalog>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model({
    catalogs: types.map(Catalog)
  })

  .actions((self) => ({
    add(item: ICatalog) {
      self.catalogs.put({ id: item.id, name: item.name, type: item.type });
    },
    clearSelected() {
      self.catalogs.forEach((c) => {
        c.selected = false;
      });
    }
  }))

  .views((self) => ({
    get items() {
      return Array.from(self.catalogs.values());
    },

    get selected() {
      const list = new Set();
      self.catalogs.forEach((c: ICatalog) => {
        if (c.selected) {
          list.add(c.id);
        }
      });

      return list;
    }
  }));
