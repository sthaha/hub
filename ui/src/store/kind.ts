import { types, Instance } from 'mobx-state-tree';

export const Kind = types
  .model({
    name: types.identifier,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type IKind = Instance<typeof Kind>;
export type IKindStore = Instance<typeof KindStore>;

export const KindStore = types
  .model({
    kinds: types.map(Kind)
  })

  .actions((self) => ({
    add(item: string) {
      self.kinds.put({ name: item, selected: false });
    },

    clearSelected() {
      self.kinds.forEach((k) => {
        k.selected = false;
      });
    }
  }))

  .views((self) => ({
    get items() {
      return Array.from(self.kinds.values());
    },

    get selected() {
      const list = new Set();
      self.kinds.forEach((c: IKind) => {
        if (c.selected) {
          list.add(c.name);
        }
      });

      return list;
    }
  }));
