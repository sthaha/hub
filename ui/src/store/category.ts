import { types, getEnv, flow, Instance } from 'mobx-state-tree';
import { Api } from '../api';

export const Tag = types.model('Tags', {
  id: types.identifierNumber,
  name: types.string
});

export const Category = types
  .model('Category', {
    id: types.identifierNumber,
    name: types.string,
    tags: types.array(types.reference(Tag)),
    selected: false
  })

  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ITag = Instance<typeof Tag>;
export type ICategory = Instance<typeof Category>;
export type ICategoryStore = Instance<typeof CategoryStore>;

export const CategoryStore = types
  .model('CategoryStore', {
    items: types.map(Category),
    tags: types.optional(types.map(Tag), {}),
    isLoading: true,
    err: ''
  })

  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    },

    get count() {
      return self.items.size;
    },

    get values() {
      return Array.from(self.items.values());
    },

    get selectedTags() {
      return new Set(
        Array.from(self.items.values())
          .filter((c: ICategory) => c.selected)
          .reduce((acc: number[], c: ICategory) => [...acc, ...c.tags.map((t: ITag) => t.id)], [])
      );
    }
  }))

  .actions((self) => ({
    add(item: ICategory) {
      self.items.put(item);
    },

    setLoading(l: boolean) {
      self.isLoading = l;
    },

    clearSelected() {
      self.items.forEach((c) => {
        c.selected = false;
      });
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);
        const { api } = self;

        const json = yield api.categories();

        // adding the tags to the store - normalized
        const tags: ITag[] = json.data.flatMap((item: ICategory) => item.tags);

        tags.forEach((t) => self.tags.put(t));

        // creating the model only after the store has the tags normalized
        const categories: ICategory[] = json.data.map((c: ICategory) => ({
          id: c.id,
          name: c.name,
          tags: c.tags.map((tag: ITag) => tag.id)
        }));

        categories.forEach((c: ICategory) => self.add(c));
      } catch (err) {
        self.err = err.toString();
      }

      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }));
