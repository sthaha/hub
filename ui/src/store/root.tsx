import React, { ReactChild, ReactChildren } from 'react';
import { types, getEnv, Instance } from 'mobx-state-tree';
import { CategoryStore } from './category';
import { ResourceStore } from './resource';
import { Hub, Api } from '../api';

export const Root = types.model('Root', {}).views((self) => ({
  get api(): Api {
    return getEnv(self).api;
  },
  get categories() {
    return getEnv(self).categories;
  },
  get resources() {
    return getEnv(self).resources;
  }
}));

export type IRoot = Instance<typeof Root>;

let RootContext: React.Context<IRoot>;
export const useMst = () => React.useContext(RootContext);

export const initRootStore = (api: Api) => {
  const categories = CategoryStore.create({}, { api });
  const resources = ResourceStore.create({ catalogs: {}, kinds: {} }, { api });
  return Root.create({}, { api, categories, resources });
};

interface Props {
  children: ReactChild | ReactChildren;
}

export const createProviderAndStore = (api?: Api) => {
  const root = initRootStore(api || new Hub());
  RootContext = React.createContext(root);

  const Provider = ({ children }: Props) => (
    <RootContext.Provider value={root}> {children} </RootContext.Provider>
  );
  return { Provider, root };
};

export const createProvider = (api?: Api) => {
  const { Provider } = createProviderAndStore(api);
  return Provider;
};
