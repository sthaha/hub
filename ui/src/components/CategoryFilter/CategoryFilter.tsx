import React from 'react';
import { useObserver } from 'mobx-react';
import Filter from '../Filter/Filter';
import { useMst } from '../../store/root';

const CategoryFilter: React.FC = () => {
  const { categories } = useMst();
  return useObserver(() => <Filter store={categories} header="Categories" />);
};

export default CategoryFilter;
