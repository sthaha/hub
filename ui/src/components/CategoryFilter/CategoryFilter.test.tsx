import React from 'react';
import { mount } from 'enzyme';
import CategoryFilter from './CategoryFilter';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
import { when } from 'mobx';
import { Checkbox } from '@patternfly/react-core';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider, root } = createProviderAndStore(api);

describe('CategoryFilter', () => {
  it('finds the filter component and matches the count', (done) => {
    const component = mount(
      <Provider>
        <CategoryFilter />
      </Provider>
    );

    const { categories } = root;
    when(
      () => {
        return !categories.isLoading;
      },
      () => {
        setTimeout(() => {
          expect(categories.count).toBe(5);
          component.update();

          const cf = component.find(CategoryFilter);
          expect(cf.length).toEqual(1);

          expect(component.debug()).toMatchSnapshot();

          expect(component.find(Checkbox).length).toEqual(5);

          done();
        }, 0);
      }
    );
  });
});
