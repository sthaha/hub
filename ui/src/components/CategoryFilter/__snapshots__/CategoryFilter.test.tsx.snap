// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter finds the filter component and matches the count 1`] = `
"<Provider>
   
  <CategoryFilter>
    <Filter store={{...}} header=\\"Categories\\">
      <div className=\\"Filter\\">
        <Grid sm={6} md={4} lg={3} xl2={1}>
          <div className=\\"pf-l-grid pf-m-all-6-col-on-sm pf-m-all-4-col-on-md pf-m-all-3-col-on-lg pf-m-all-1-col-on-2xl\\">
            <GridItem className=\\"Text-Header\\" span={1} rowSpan={2}>
              <div className=\\"pf-l-grid__item pf-m-1-col pf-m-2-row Text-Header\\">
                <Text component=\\"h1\\" style={{...}}>
                  <h1 style={{...}} data-pf-content={true} className=\\"\\">
                    Categories
                  </h1>
                </Text>
              </div>
            </GridItem>
            <GridItem rowSpan={2}>
              <div className=\\"pf-l-grid__item pf-m-2-row\\">
                <Button variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res]}>
                  <button onClick={[Function: res]} aria-disabled={false} aria-label=\\"Clear\\" className=\\"pf-c-button pf-m-plain\\" disabled={false} tabIndex={[undefined]} type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe={true} data-ouia-component-id={1}>
                    <TimesIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                      <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 352 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                        <path d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" transform=\\"\\" />
                      </svg>
                    </TimesIcon>
                  </button>
                </Button>
              </div>
            </GridItem>
          </div>
        </Grid>
        <Grid>
          <div className=\\"pf-l-grid\\">
            <GridItem className=\\"Checkboxes\\">
              <div className=\\"pf-l-grid__item Checkboxes\\">
                <Checkbox label=\\"Build Tools\\" isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"1\\" name=\\"Build Tools\\" className=\\"\\" isValid={true} isDisabled={false}>
                  <div className=\\"pf-c-check\\">
                    <input id=\\"1\\" name=\\"Build Tools\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"1\\">
                      Build Tools
                    </label>
                  </div>
                </Checkbox>
                <Checkbox label=\\"CLI\\" isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"2\\" name=\\"CLI\\" className=\\"\\" isValid={true} isDisabled={false}>
                  <div className=\\"pf-c-check\\">
                    <input id=\\"2\\" name=\\"CLI\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"2\\">
                      CLI
                    </label>
                  </div>
                </Checkbox>
                <Checkbox label=\\"Cloud\\" isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"3\\" name=\\"Cloud\\" className=\\"\\" isValid={true} isDisabled={false}>
                  <div className=\\"pf-c-check\\">
                    <input id=\\"3\\" name=\\"Cloud\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"3\\">
                      Cloud
                    </label>
                  </div>
                </Checkbox>
                <Checkbox label=\\"Deploy\\" isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"4\\" name=\\"Deploy\\" className=\\"\\" isValid={true} isDisabled={false}>
                  <div className=\\"pf-c-check\\">
                    <input id=\\"4\\" name=\\"Deploy\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"4\\">
                      Deploy
                    </label>
                  </div>
                </Checkbox>
                <Checkbox label=\\"Image Build\\" isChecked={false} onChange={[Function: onChange]} aria-label=\\"controlled checkbox\\" id=\\"5\\" name=\\"Image Build\\" className=\\"\\" isValid={true} isDisabled={false}>
                  <div className=\\"pf-c-check\\">
                    <input id=\\"5\\" name=\\"Image Build\\" className=\\"pf-c-check__input\\" type=\\"checkbox\\" onChange={[Function (anonymous)]} aria-invalid={false} aria-label=\\"controlled checkbox\\" disabled={false} checked={false} />
                    <label className=\\"pf-c-check__label\\" htmlFor=\\"5\\">
                      Image Build
                    </label>
                  </div>
                </Checkbox>
              </div>
            </GridItem>
          </div>
        </Grid>
      </div>
    </Filter>
  </CategoryFilter>
   
</Provider>"
`;
